internal/config/config.go

```diff
@@ -509,6 +509,75 @@ func Load(cfg *Config) *viper.Viper {
 	return v
 }
 
+// Log logs the configuration
+func (c *Config) Log(logger slog.Logger) {
```

Comment:
I suggest to divide the method in smaller private ones, so this method could looks like:

```go
func (c *Config) Log(logger slog.Logger) {
	if c == nil {
		panic("'c' is a nil &Config{} pointer")
	}
	logger.Info(
		"Configuration",
		slog.Bool("Loaded", c.Loaded),
		c.logGroupWeb(&c.Web),
		c.logGroupDatabase(&c.Database),
		c.logGroupLogging(&c.Logging),
		c.logGroupMetrics(&c.Metrics),
		c.logGroupClients(&c.Clients),
		c.logGroupApplication(&c.Application),
		// No logGroupSecrets() is intentional
	)
}
```

And for instance for the group web could looks like:

```go
func (c *Config) logGroupWeb(data *Web) slog.Attr {
	return slog.Group("Web",
		slog.Int("Port", int(data.Port)),
	)
}
```

In this way, the unit tests would be easier to accommodate and independent of the format; we could check for every private method, that the `slog.Attr` structure returned fit our expectations which is independent of the logging format; for instance for logGroupWeb it could looks like:

```go
func TestGroupWeb(t *testing.T) {
	data := &Config{
		Web: Web{
			Port: 8000,
		},
	}
	assert.Equal(t,
		slog.Group("Web",
			slog.Int("Port", 8000),
		),
		data.logGroupWeb(&data.Web),
	)
}
```

In that way we could cover more lines with the unit tests, without consider the output format complexity of the log.

For the log format, I had a vague memory in the past; I have found this unit test, but I agree that check the output format as unnecessary:

https://github.com/podengo-project/idmsvc-backend/blob/main/internal/infrastructure/logger/meta_handler_test.go#L122

> Dividing the method in smaller one, we could get coverage for almost everything without enter on the output format details.

Reply:
I don't see a lot of benefits in such test as it is practically copying the implementation and thus would break with small implementation change - i.e. it is testing implementation and not the behavior.

Did a small research and I've added a test that verifies at least one attr from eah section + that all secrets values are not present.